simulationworkflowschema/general.py

```diff
@@ -231,24 +289,232 @@ def normalize(self, archive, logger):
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=calculation.method_ref)
+                    )
                 elif len(self._calculations) == len(self._methods):
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=self._methods[n]))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=self._methods[n])
+                    )
                 elif len(self._methods) == 1:
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=self._methods[0]))
-                self.tasks.append(Task(name=f'Step {n}', inputs=inputs, outputs=outputs))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=self._methods[0])
+                    )
+                self.tasks.append(
+                    Task(name=f"Step {n}", inputs=inputs, outputs=outputs)
+                )
+
+
+class BeyondDFT(SerialSimulation):
+    """
+    Base class used to normalize standard workflows beyond DFT containing two specific
+    SinglePoint tasks (GWWorkflow = DFT + GW, DMFTWorkflow = DFT + DMFT,
+    MaxEntWorkflow = DMFT + MaxEnt, and so on) and store the outputs in the self.results
+    section.
+    """
+
+    def _resolve_outputs_section(self, output_section, task: TaskReference) -> None:
+        """
+        Resolves the output_section of a task and stores the results in the output_section.
+
+        Args:
+            task (TaskReference): The task from which the outputs are got.
+        """
+        for name, section in output_section.m_def.all_quantities.items():
+            name = f"{name}_electronic" if name in ["dos", "band_structure"] else name
+            try:
+                calc_section = getattr(task.outputs[-1].section, name)
+                if calc_section:
+                    output_section.m_set(section, calc_section)
+            except Exception:
+                continue
+
+    def get_electronic_structure_workflow_results(self, task_map: dict) -> None:
+        """
+        Gets the standard electronic structure workflow results section by resolving the
+        outputs specified in the `task_map`.
+
+        Args:
+            task_map (dict): The dictionary used to resolve the outputs sections.
+        """
+        for method, task in task_map.items():
+            outputs = ElectronicStructureOutputs()
+            self._resolve_outputs_section(outputs, task)
+            setattr(self.results, f"{method}_outputs", outputs)
+
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+        if len(self.tasks) != 2:
+            logger.error("Expected two tasks.")
+            return
+
+        # We extract the workflow name from the tasks names
+        workflow_name = []
+        for task in self.tasks:
+            workflow_name.append(task.name)
+        self.name = "+".join(workflow_name)
```

Comment:
this could be problematic if task.name is None. do '+'.join([task.name for task in self.tasks if task.name])